ci: twister: blackbox: do not fetch hal modules

nashif opened the pull request:
Do not fetch optional modules while running this action.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: twister: blackbox: do not fetch optional modules

Do not fetch optional modules while running this action.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

fabiobaltieri commented on .github/workflows/twister_tests_blackbox.yml:
```diff
@@ -43,6 +43,7 @@ jobs:
         echo "$HOME/.local/bin" >> $GITHUB_PATH
 
         west init -l . || true
+        west config manifest.group-filter -- +ci,-optional
```
`-optional` is redundant right? that should be the default

not asking to change it just trying to understand

nashif replied:
> `-optional` is redundant right? that should be the default

is it the default? defined where? AFAIK, without filters, everything is fetched by default. see `.github/workflows/compliance.yml` for example.

Either way, this fix is not enough, we have a "bigger" problem....

fabiobaltieri replied:
> is it the default? defined where?

Here? https://github.com/zephyrproject-rtos/zephyr/blob/main/west.yml#L27 if that's not effective you want to filter out babblesim as well

nashif replied:
oh yeah, you are right. Then I have to find a more directy approach now :(

nashif committed:
twister: tests: use platforms that do not depend on a HAL

Do not depend on platforms that need a HAL. This should speed things up
and should resolve issues where runner did not have enough space to deal
with all HALs.
t
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
manifest: add mcuboot to bootloader group

Add mcuboot to bootloader group to allow filter where needed.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
